win32/agent.py

```diff
@@ -127,10 +128,15 @@ class ProcessWatchDog(object):
     Monitor the attached process.
     Restarts when it exits until the limit set is reached.
     """
+    _RESTART_TIMEFRAME = 3600
+
     def __init__(self, name, process, max_restarts=5):
+        """
+        :param max_restarts: maximum number of restarts per _RESTART_TIMEFRAME timeframe.
+        """
         self._name = name
         self._process = process
-        self._count_restarts = 0
+        self._restarts = deque([])
         self._MAX_RESTARTS = max_restarts
```

Comment:
I don't really like this way (not your PR, but close), having a "dynamic" constant. Maybe you could instead have a `DEFAULT_MAX_RESTARTS = 5`, `max_restarts=None`, and `self._max_restarts = max_restarts or self.DEFAULT_MAX_RESTARTS`: what do you think ?